### ISSUE TITLE
Animations should be local to scene by default

### ISSUE THREAD

**Issue by nobuyukinyuu** [godotengine/godot#30197]
**Godot version:**
3.1.1

**Issue description:**
When instancing a scene which inherits from a scene with an `AnimationPlayer`, any changes to animations and animation tracks will affect **all** instances, including deleting animations.  This also applies to copied-and-pasted animations and tracks across disparate scenes.  

This referencing behavior may be considered unexpected to users, as the default behavior of other node properties is typically local-but-inherited and the `Animation`s' appearance as a reference is obfuscated from the inspector view without explicitly opening it from the animation tab.  This behavior also occurs when copying and pasting animations across non-inherited scenes, including the deletion of whole `Animation`s and tracks.

**Comment by TheDuriel** [comment]
Resources are shared by default to avoid unnecessary duplication. Modifying animations at runtime is not something everyone needs, so the vast majority of animations will never need duplicating. I don't think this is a good idea.

You can instead perhaps save your animations to disk, duplicate those files. And load them as unique instances as needed.

**Comment by avencherus** [comment]
It could benefit from some clarity.  The concept of Resources and how the AnimationPlayer manages them is not indicated.  I'm still never sure exactly when the Animation has been duplicated or is referenced.

As it is currently, there is a legitimate issue with accidentally destroying work when dealing with Animations.

**Comment by TheDuriel** [comment]
Resources, an thus Animations are _never_ duplicated unless you do so manually.

@avencherus 

Resources are indeed in need to a proper docs page. They are 50% of Godots core architecture (together with Nodes), but few people are aware of this. (Resources can do a lot of cool stuff actually.)

**Comment by nobuyukinyuu** [comment]
@TheDuriel  

I never mentioned this being a runtime issue.  This is a design time issue, as indicated with my concerns with how the inspector doesn't make it obvious that an animation is a typical ByRef resource unless you explicitly access it from the animation tab's pulldown menu.  Animations themselves perhaps should be visible in the AnimationPlayer's inspector context in the same way as other resources are, with the same context menus.

The problem comes when trying to copy animations over to a seperate player, or alter properties of an AnimationPlayer inherited from a base node *at design time*.  Most users will treat the fields of the animation tab context as overridable properties of the inherited instance of the scene, not as references to the base animation.

**Comment by nobuyukinyuu** [comment]
The way animations are currently stored in an `AnimationPlayer` is currently done through functions and isn't exposed as a property.  I'm not sure how the context for this node type works, but some sort of exposed access to the animation list via the inspector seems like the way to go, the less "customized" hacking to the context necessary, the better, since I'm sure there's going to be someone arguing that this is unnecessary duplicate functionality.

Mockup of one possible solution  (Shows explicitly in the inspector that Animations are References, important for design-time inherited scenes):
![image](https://user-images.githubusercontent.com/1023003/60399458-239e0100-9b2a-11e9-8fa4-936d03ceba0a.png)

Mockup of another (Addresses copying and pasting of Animations in an explicit way as well as allows for duplicate animations to be copied across `AnimationPlayer`s without the hassle of having to save the Resource to an external file):
![image](https://user-images.githubusercontent.com/1023003/60399622-9a3bfe00-9b2c-11e9-97f8-6d95294a3764.png)

Clarification can be further improved by appending resource ID context to the animation name in the dropdown, similar to how Godot 3.0 showed a resource number in the inspector context.  In both examples, the second animation was made with copy and paste, the third, with Duplicate, with "expected" results:
![image](https://user-images.githubusercontent.com/1023003/60399739-d83a2180-9b2e-11e9-8057-207c0528c729.png)

**Comment by ChildLearningClub** [comment]
New users to Godot including myself have been caught by this, probably hard to find anyone who hasn’t unless you know prior to using the engine how it works. I understand now that it is to “save on resources” :) ( Sorry had to put that in there, not that anyone’s laughing right now over all their over writing changes)

I like Godot’s built in efficiency. What’s also nice is that the referencing is abstracted away making that built in efficiency automatic. So why not keep that how it is and when a user goes to edit / modify a referenced object ask the user if they would like to “Keep as Reference” . `“You are about to edit a referenced object, which will change the original. Would you like to “Continue” / “Continue and Keep as Reference” (Button) or “Copy to New Animation” (Button).` This should at least take care of the confusion upfront, having the “appended resource ID information to the animation name” as mentioned above would also be helpful in tracking down which animation is being used, as that newly created object can now be used as a reference elsewhere.

So to summarize objects are copied as reference, until edited, at which point the user is asked if they want to keep as reference or copy to new animation.

**Comment by AThousandShips** [comment]
It isn't a referenced object though, all resources are accessed as references, just some are unique, there's no original or reference, all are references of each other

**Comment by ChildLearningClub** [comment]
@AThousandShips, thank you for explaining. I guess I thought that it would be possible to simply have some kind of message displayed when an object is edited or edited object about to be saved that would indicate that it would change other objects also attached to that resource? Or that it’s not “unique”? I hope I’m getting the terminology correct?

**Comment by AThousandShips** [comment]
That would absolutely be possible! I don't know if it wouldn't just annoy people who want to have the resource reused, as is often the case

**Comment by jeanmarc** [comment]
Godot Noob here, but just wanted to suggest something. Blender shows a reference count in its editors for objects/resources that are being used in multiple places (like materials), and provides a means to make them unique by clicking on the reference count. That could be a minimally intrusive way of showing the multiple-use situation and offering a way to make-unique?
